File: practice-algos/counting_valleys/gpt_sol.js
function countingValleys(steps, path) {
    let seaLevel = 0;
    let valleys = 0;
    let inValley = false;

    for (let step of path) {
        if (step === 'U') {
            seaLevel++;
        } else {
            seaLevel--;
        }

        if (seaLevel < 0 && !inValley) {
            valleys++;
            inValley = true;
        } else if (seaLevel >= 0) {
            inValley = false;
        }
    }

    return valleys;
}

// Example usage
const steps = 8;
const path = "DDUUUUDD";
console.log(countingValleys(steps, path));  // Output: 1

File: practice-algos/counting_valleys/gpt_sol.py
def countingValleys(steps, path):
    sea_level = 0
    valleys = 0
    in_valley = False

    for step in path:
        if step == 'U':
            sea_level += 1
        else:
            sea_level -= 1

        if sea_level < 0 and not in_valley:
            valleys += 1
            in_valley = True
        elif sea_level >= 0:
            in_valley = False

    return valleys

# Example usage
steps = 8
path = "DDUUUUDD"
print(countingValleys(steps, path))  # Output: 1
